### PR TITLE
Improve dump-signedexchange tests

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
Check that cert-url and validity-url have a 200 status code, and have the expected content-type.

### CODE REVIEW

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 1]
```diff
@@ -34,31 +72,65 @@ export class SxgDumpSignedExchangeVerify extends Rule {
...
-    } else {
-      return this.pass();
     }
+    return this.pass();
   }
   meta() {
     return {
       url: "",
-      title: "dump-signedexchange -verify does not report errors",
+      title: "no SXG errors found",
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: The error message: "FAIL no SXG errors found" is a bit confusing. Maybe: "valid SXG"?

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 2]
```diff
@@ -34,31 +72,65 @@ export class SxgDumpSignedExchangeVerify extends Rule {
...
+    if (!validityUrl) {
+      return this.fail(`Can't find valid [cert-url] [${JSON.stringify(sxg)}]`);
+    }
+    if (!(await urlHasContentType(validityUrl, headers, "application/cbor"))) {
+      return this.fail(
+        `validity-url [${validityUrl}] is not found or has wrong content type`
+      );
+    }
+    const expires = sxg.signatures[0]["Params"]["expires"] as number;
+    if (7 * 24 * 60 * 60 + Date.now() / 1000 < expires) {
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: magic number, please replace with a constant

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 3]
```diff
@@ -4,20 +4,57 @@ import { fetchToCurl } from "../helper";
...
+}
+
+const REQUEST_HEADERS = {
+  "accept": "application/signed-exchange;v=b3",
+  "amp-cache-transform": `google;v="1"`
+};
+const EXPECTED_VERSION = "1b3";
+
 export class SxgDumpSignedExchangeVerify extends Rule {
   async run({ url, headers }: Context) {
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: suggestion: split the method up into multiple methods, where each method implements a different check. Currently, it's very hard to determine from the code what is checked.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 4]
```diff
@@ -26,6 +63,7 @@ export class SxgDumpSignedExchangeVerify extends Rule {
     const body = await res.buffer();
     const CMD = `dump-signedexchange`;
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: move constant to head

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 5]
```diff
@@ -26,6 +63,7 @@ export class SxgDumpSignedExchangeVerify extends Rule {
...
     const ARGS = [`-verify`, `-json`];
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: move constant to head

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts [hunk 6]
```diff
@@ -26,6 +63,7 @@ export class SxgDumpSignedExchangeVerify extends Rule {
...
+    const DEBUG = `${fetchToCurl(url, opt, false)} | ${CMD} ${ARGS.join(" ")}`;
```
Comment (sebastianbenz) [inline packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: nit: lowercase as it's not a constant.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 4 above)
Comment (sebastianbenz) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: suggestion: rename also to CMD_DUMP_SXG

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 1 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: Changed to `verification by ${CMD_DUMP_SXG} tool`.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 2 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: Now `7 * SECONDS_IN_A_DAY`. Or you want the whole thing a constant?

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 3 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: I had a few goes and couldn't figure out a way to improve it (with the current reporting structure). Have added in some whitespace to create sections and will think a bit more about it. I'm not happy with this either.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 4 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: Done.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 5 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: Done.

File: packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts (hunk 6 above)
Comment (ithinkihaveacat) [reply packages/linter/src/rules/SxgDumpSignedExchangeVerify.ts]: Done.

**Comment by ithinkihaveacat** [comment]
@sebastianbenz PTAL. I think all comments resolved with the exception of breaking the method into pieces, which I agree is needed but I'd rather not do in this PR.

**Comment by sebastianbenz** [comment]
@ithinkihaveacat can you take a look at the test failure

**Comment by ithinkihaveacat** [comment]
The test failure seems to be network related (and unrelated to this PR)? Although restarting it hasn't helped. Will have another look in a bit.

**Comment by sebastianbenz** [comment]
works for me as well

**Pull request merged**